ocaml/xapi/stream_vdi.ml

```diff
@@ -100,6 +100,53 @@ let write_block ~__context filename buffer ofd len =
     then raise (Api_errors.Server_error (Api_errors.client_error, [ExnHelper.string_of_exn e]))
     else raise e
 
+(* Example = [{"length": 2097152, "flags": 0}, {"length": 1902848, "flags": 3}]*)
+type descriptor = {
+  length : int64;
+  status_flags : int32;
+}
+
+let cycle_descriptors descriptor_list offset =
+  let rec range a b =
+    if a > b then []
+    else a :: range (a + 1) b
+  in
+  (* Insert proper not_empty function here *)
+  let not_empty status_flag =
```

Comment:
Unless you expect more cases to come up, this could be simplified:
```
let not_empty status_flag =
  Int32.to_int status_flag = 3
```

Reply:
@gaborigloi mentioned that there is some `Nbd.Protocol.AllocationMetadataFlag.of_int32` function we can use in the final implementation so might leave as is for now.

Reply:
It's empty if either bit 0 or 1 is set: https://github.com/xapi-project/vhd-tool/blob/master/src/nbd_input.ml#L32
So we need to slightly change this

Reply:
And we can't use that `Nbd` function instead, because that will get merged later, it's in the nice to have epic.

Reply:
@gaborigloi so Christian's suggestion is valid as there is only one options for non-empty?

Reply:
If either of these bits is set, the block is empty.
See NBD_STATE_HOLE and NBD_STATE_ZERO in the NBD protocol docs: https://github.com/NetworkBlockDevice/nbd/blob/master/doc/proto.md
So we need to check for the presence of these bits in the status_flag(s) integer

Reply:
I think we need to `or` it with `0b0011`, because if *either* of these bits is set, the extent is empty.